app/models/concerns/article.rb

```diff
@@ -37,7 +37,7 @@ def set_user
 
   def set_channel
     return if self.channel.present?
-    if User.current && User.current.is_a?(BotUser)
+    if self.user.is_a?(BotUser) || User.current && User.current.is_a?(BotUser)
```

Comment:
@caiosba here we were only checking `User.current`. Should we also check `self.user`?

Reply:
I think we should only check for `self.user`. Rationale: We already have a `set_user` callback before this one that sets the user based on `User.current`.